### PR TITLE
explorer: add dark theme

### LINKED ISSUE #1978: Add dark theme
#### Summary
Implement the dark theme design.

### PR DESCRIPTION
**Pull request by deuch13**
Resolves #1978

### CODE REVIEW

File: explorer/src/lib/stores/appStore.js [hunk 1]
```diff
@@ -32,8 +43,17 @@ const setNetwork = (network) =>
...
   });
 
+/** @param {boolean} darkMode */
+const setTheme = (darkMode) => {
+  set({
+    ...get(store),
+    darkMode,
+  });
+};
+
```
Comment (nortonandreev) [inline explorer/src/lib/stores/appStore.js:53]: If we don't have a switch, I would propose that we remove this for the time-being.

**Comment by nortonandreev** [comment]
**Feedback on Button Variants and Theming:**

The button variant should stay consistent across themes, with only the appearance changing in dark mode. This maintains semantic meaning, simplifies the code, and ensures a consistent user experience. Use CSS variables to adjust the appearance, like in the wallet application where `--control-bg-color` changes from `--smokey-black` in light mode to `--light-grey` in dark mode. This approach keeps the UI clear and maintainable.

File: explorer/src/lib/components/data-card/DataCard.svelte [hunk 2]
```diff
@@ -37,7 +39,9 @@
       <Button
         on:click={headerButtonDetails.action}
         text={headerButtonDetails.label}
-        variant={headerButtonDetails.variant || "secondary"}
+        variant={darkMode
+          ? "tertiary"
+          : headerButtonDetails.variant || "secondary"}
```
Comment (nortonandreev) [inline explorer/src/lib/components/data-card/DataCard.svelte:44]: Check my comment on the button variants and theming.

The main idea is we should not change the variant semantically based on the theme. Instead, we should tweak the appearance of the variant to match the dark theme, as variants like "primary", "secondary", "danger", etc., are defined based on their roles and importance in the UI, not their visual appearance.

File: explorer/src/lib/components/navbar/Navbar.svelte [hunk 3]
```diff
@@ -60,7 +60,9 @@
     showSearchNotification = false;
   });
 
-  $: ({ networks } = $appStore);
+  $: ({ darkMode, networks } = $appStore);
+
+  const isDarkMode = darkMode;
```
Comment (nortonandreev) [inline explorer/src/lib/components/navbar/Navbar.svelte:65]: You should be able to directly bind to the `darkMode` from the appStore, I don't think this constant is needed.

File: explorer/src/lib/components/navbar/Navbar.svelte [hunk 4]
```diff
@@ -114,8 +116,17 @@
...
         />
       {/if}
+
+      <!-- No design implementation -->
+      <!-- <Switch
+        bind:value={isDarkMode}
+        on:change={() => {
+          appStore.setTheme(isDarkMode);
+        }}
+      /> -->
```
Comment (nortonandreev) [inline explorer/src/lib/components/navbar/Navbar.svelte:129]: Again, let's remove that, as we are not going to have a switch for the time-being.

File: explorer/src/lib/components/search-notification/SearchNotification.svelte [hunk 5]
```diff
@@ -31,6 +34,7 @@
         position: "after",
         size: "normal",
       }}
+      variant={darkMode ? "tertiary" : "secondary"}
```
Comment (nortonandreev) [inline explorer/src/lib/components/search-notification/SearchNotification.svelte:37]: See the comment about the variants.

**Comment by nortonandreev** [comment]
https://github.com/user-attachments/assets/d3a682b7-0731-4bd0-ba87-91078d0afa62

I have a feeling that the hover state should be something else? 👀

File: explorer/src/lib/stores/appStore.js (hunk 1 above)
Comment (deuch13) [reply explorer/src/lib/stores/appStore.js:52]: One reason I would not remove this is as a helper for tests, as there is no update method on the store.

**Comment by deuch13** [comment]
> Screen.Recording.2024-07-17.at.12.43.46.mov 
> I have a feeling that the hover state should be something else? 👀

I've addressed this by bringing in the design we currently have in the wallet.

**Comment by deuch13** [comment]
> **Feedback on Button Variants and Theming:**
> 
> The button variant should stay consistent across themes, with only the appearance changing in dark mode. This maintains semantic meaning, simplifies the code, and ensures a consistent user experience. Use CSS variables to adjust the appearance, like in the wallet application where `--control-bg-color` changes from `--smokey-black` in light mode to `--light-grey` in dark mode. This approach keeps the UI clear and maintainable.

Agreed, this has been fixed by bringing in the update button and language css from the wallet app.

File: explorer/src/lib/stores/appStore.js (hunk 1 above)
Comment (ascartabelli) [reply explorer/src/lib/stores/appStore.js:52]: > One reason I would not remove this is as a helper for tests, as there is no update method on the store.

That is easily fixed by mocking what `matchMedia` returns, but I would keep it nonetheless.

**Review by kieranhall — APPROVED** [review]
LGTM

**Pull request merged**